Api/Src/Services/AuthService.ts

```diff
@@ -0,0 +1,70 @@
+import {JWT_TOKEN_EXPIRESIN} from "../../Constants";
+import {UserModel} from "../Models/User";
+import { UserType} from "../Models/User";
+import * as mongoose from 'mongoose';
+import UserRepository from "../Repositories/User";
+import * as bcrypt from "bcrypt";
+const jwt = require('jsonwebtoken');
+
+type User = UserModel & mongoose.Document;
+
+export default class AuthService {
+    private saltRounds: number = 10;
+
+    constructor(private repository: UserRepository) { }
+
+    generateToken = (user: User): string => {
+        return jwt.sign(
+            { _id: user._id, type: user.type },
+            process.env.JWT_PRIVATE_KEY,
```

Comment:
It has been resolved in container as appContainer.declare("jwtKey", (c) => process.env.JWT_PRIVATE_KEY); so maybe it's better to pass it through constructor.

Reply:
Yea you're definitely right, there is no point in reaching it in another way. 
I moved also JWT_TOKEN_EXPIRESIN=20m to env and injected through constructor.